src/BuildScriptGenerator/DefaultDockerfileGenerator.cs

```diff
@@ -255,5 +261,49 @@ private string ConvertToRuntimeVersion(string platformName, string platformVersi
         {
             return this.platformDetector.GetCompatiblePlatforms(ctx);
         }
+
+        private void ValidateDockerfileImages(DockerfileProperties dockerfileProperties)
+        {
+            if (string.IsNullOrEmpty(dockerfileProperties.BuildImageName))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Provided build image name parsed from the provided --build-image argument is empty. " +
+                    "Please provide a valid value for --build-image, or remove the argument to use the default " +
+                    "'cli:stable' build image from mcr.microsoft.com/oryx.");
+            }
+
+            if (string.IsNullOrEmpty(dockerfileProperties.BuildImageTag))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Provided build image tag parsed from the provided --build-image argument is empty. " +
+                    "Please provide a valid value for --build-image, or remove the argument to use the default " +
+                    "'cli:stable' build image from mcr.microsoft.com/oryx.");
+            }
+
+            if (string.IsNullOrEmpty(dockerfileProperties.RuntimeImageName))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Either the value provided to the --runtime-platform argument is empty, or the platform " +
+                    "discovered by Oryx does not have any available or supported runtime images. Please view the following " +
+                    "document for more information on runtimes supported by Oryx: " +
+                    "https://github.com/microsoft/Oryx/blob/main/doc/supportedRuntimeVersions.md");
```

Comment:
Nit: please consider using an aka.ms link instead, so that if in the future the location of this doc changes, all the older images don't redirect to a wrong URL.

Reply:
Good call -- https://aka.ms/oryx-runtime-images